widgets/pagination/__tests__/pagination-test.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-env mocha */
+
+import React from 'react';
+import expect from 'expect';
+import sinon from 'sinon';
+import jsdom from 'mocha-jsdom';
+
+import toEqualJSX from 'expect-to-equal-jsx';
+expect.extend({toEqualJSX});
+
+import pagination from '../pagination';
+import Pagination from '../../../components/Pagination/Pagination';
+
+describe('pagination()', () => {
+  jsdom();
+
+  var ReactDOM;
+  var container;
+  var widget;
+  var results;
+  var helper;
+
+  beforeEach(() => {
+    ReactDOM = {render: sinon.spy()};
+    pagination.__Rewire__('ReactDOM', ReactDOM);
+    pagination.__Rewire__('autoHide', sinon.stub().returns(Pagination));
+
+    container = document.createElement('div');
+    widget = pagination({container, scrollTo: false});
+    results = {hits: [{first: 'hit', second: 'hit'}], nbHits: 200, hitsPerPage: 10, nbPages: 20};
+    helper = {
+      setCurrentPage: sinon.spy(),
+      search: sinon.spy()
+    };
+  });
+
+  it('configures nothing', () => {
+    expect(widget.getConfiguration).toEqual(undefined);
+  });
+
+  it('sets the page', () => {
+    widget.setCurrentPage(helper, 42);
+    expect(helper.setCurrentPage.calledOnce).toBe(true);
+    expect(helper.search.calledOnce).toBe(true);
```

Comment:
> If the implementation change and doesn't require to call search anymore, we would need to change this test.

But that's the good thing from this test. If we remove the .search() call from the code this will fail and we will have to think